tests/testsuite/artifact_dep.rs

```diff
@@ -1445,6 +1445,55 @@ foo v0.0.0 ([CWD])
         )
         .run();
 }
+
+#[cargo_test]
+fn artifact_dep_target_specified() {
```

Comment:
This test passes even without the patch in `ops::tree::graph`. We should probably first create a test verifying the current panic behavior in a commit, followed by the other commit fixing both the behavior and the test.